### ISSUE TITLE
Refresh error while refactoring names

### ISSUE THREAD

**Issue by jcoeltjen** [eclipse/xtext-xtend#166]
When renaming a constant that resides outside test from inside a test a refresh error occurs.

Preparations:

1. Create new Xtext Project
2. Write a Validator and with at least one public String constant 
3.  Use the String constant in the unit tests.

Reproduce the error:

1.  Rename the String constant from your validator class from the unit test.

Now the refresh fails. In the unit test the newly renamed constant will be shown as not existing, altough it has been properly renamed in the validator class.

My current workaround is to comment out the failing line in the test, wait until eclipse rebuilt the workspace and then uncomment the line again.

The expected behaviour would be that eclipse rebuilds the validator class before attempting to rebuild the test class.

**Comment by cdietrich** [comment]
could be https://github.com/eclipse/xtext-xtend/pull/161

**Comment by cdietrich** [comment]
i am wondering where there is no rebuild sheduled after renaming

**Comment by cdietrich** [comment]
jdt has the same problem, the send their events, this makes xtext beeding notified on jdt changes

**Comment by cdietrich** [comment]
maybe calling `syncUtil.waitForBuild(new NullProgressMonitor());` at the end of 
`org.eclipse.xtext.ui.refactoring.ui.RenameRefactoringExecuter.execute(IEditorPart, ProcessorBasedRefactoring)` helps

**Comment by cdietrich** [comment]
@JanKoehnlein what do you think?

**Comment by cdietrich** [comment]
```
IResource resource = editor.getAdapter(IResource.class);
		if (resource != null) {
			IProject project = resource.getProject();
			if (project != null) {
				try {
					project.build(IncrementalProjectBuilder.INCREMENTAL_BUILD, new NullProgressMonitor());
				} catch (CoreException e) {
					LOG.error("Error Building the Project", e);
					showFatalErrorMessage(editor.getSite().getShell(), "Error Building the Project.");
				}
			}
		}
```

**Comment by JanKoehnlein** [comment]
It's better to explicitly trigger a build than to wait for an auto-build.

**Comment by cdietrich** [comment]
Well both trigger an incremental build

The question is wheater we need to inc build all projects or can figure out the rooting project (editor might be the one from a downstream project)

**Comment by cdietrich** [comment]
see https://bugs.eclipse.org/bugs/show_bug.cgi?id=474781 as well

**Comment by cdietrich** [comment]
fixed in 2.12
